src/NServiceBus.Persistence.DynamoDB/Saga/SagaPersister.cs

```diff
@@ -249,18 +244,15 @@ public Task Update(IContainSagaData sagaData, ISynchronizedStorageSession sessio
 
         Dictionary<string, AttributeValue> Serialize(IContainSagaData sagaData, int version)
         {
-            // All this is super allocation heavy. But for a first version that is OK
-            var sagaDataJson = JsonSerializer.Serialize(sagaData, sagaData.GetType());
-            var doc = Document.FromJson(sagaDataJson);
-            var map = doc.ToAttributeMap();
-            map.Add(configuration.Table.PartitionKeyName, new AttributeValue { S = $"SAGA#{sagaData.Id}" });
-            map.Add(configuration.Table.SortKeyName, new AttributeValue { S = $"SAGA#{sagaData.Id}" });  //Sort key
+            var sagaDataMap = DataSerializer.Serialize(sagaData, sagaData.GetType());
+            sagaDataMap.Add(configuration.Table.PartitionKeyName, new AttributeValue { S = $"SAGA#{sagaData.Id}" });
```

Comment:
not limited to this PR, but do we have some way to prevent users from using the PK/SK names as properties in a saga data?